anvio/filesnpaths.py

```diff
@@ -368,3 +368,192 @@ def get_name_from_file_path(file_path, postfix_separator="."):
     splits = os.path.basename(os.path.abspath(file_path)).split(postfix_separator)
 
     return splits[0] if len(splits) in [1, 2] else '.'.join(splits[:-1])
+
+
+class AppendableFile:
+    """A file class for repeatedly opening the same file and appending data to it.
+
+    Parameters
+    ==========
+    file_path : string
+        the path to the file
+    append_type : data type, None
+        If you provide a data type here, then only data of that type will be allowed to be appended to the file
+    fail_if_file_exists : boolean
+        if True, throw an error upon initialization if the file already exists. If False,
+        start appending to the end of the file.
+    overwrite_if_file_exists : boolean
+        if True (and fail_if_file_exists is False), the existing file will be removed before adding new data.
```

Comment:
I would argue this functionality lies outside the responsibility of a class called `AppendableFile`

Reply:
I see your point. I wanted this functionality because when I re-run `anvi-estimate-metabolism` with the same output file prefix, I did not want those output files to be continuously appended to, I wanted to clear out the old data and then start appending. But I realize now that I can control this from the estimation function that opens the output files instead of doing it from the `AppendableFile` class. I think I will do that instead.

(🤔 this also just made me realize that not overwriting existing files will allow us to combine data from different runs of `anvi-estimate-metabolism` into the same file, which could be useful for some people. I will implement that as well, thanks for the inspiration!)